### DISCUSSION TITLE
Go back to previous state

### DISCUSSION THREAD

**Discussion by pke** [statelyai/xstate#2557] [Q&A]
I can enter a state "b" from 3 different other states and and would like to have a `BACK` event in `b` that transitions back to the previous state. 
Is this somehow possible? Using the [History](https://xstate.js.org/docs/guides/history.html#history) maybe?

**Comment by ChrisShank** [comment]
Using a history state could be what you need here. Here is an [example](https://stately.ai/viz?id=4064ae9f-c586-4e1c-ab38-06619d3ac467#). Just keep in mind that history state only keeps track of the last **state** you were in, not a stack of previous states.

To use the history state you will have to nest those three other states in another state like I did with `fanOn`.
